daffodil-io/src/main/scala/org/apache/daffodil/io/InputSource.scala

```diff
@@ -325,8 +361,10 @@ class BucketingInputSource(inputStream: java.io.InputStream, bucketSize: Int = 1
     } else {
       val (bucketIndex, byteIndex) = bytePositionToIndicies(curBytePosition0b)
 
-      if (buckets(bucketIndex.toInt) == null)
-        throw new BacktrackingException
+      if ((bucketIndex < 0) || (buckets(bucketIndex.toInt) == null)) {
+        _isValid = false
+        throw new BacktrackingException("Attempted to backtrack to position %d, which exceeds maximum backtrack length of %d".format(bucketIndex, maxCacheSizeInBytes))
```

Comment:
This says "position %d", but passes in bucketIndex as the parameter, which isn't a position, but an internal thing that really won't mean anything to a user. I think we instead want curBytePosition or something? Might also say "backtrack to byte %d" since sometimes we report bit positions.